src/plugins/discover/public/application/apps/main/utils/get_fetch_observable.ts

```diff
@@ -0,0 +1,61 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+import { merge } from 'rxjs';
+import { debounceTime, filter, tap } from 'rxjs/operators';
+
+import { FetchStatus } from '../../../types';
+import type {
+  AutoRefreshDoneFn,
+  DataPublicPluginStart,
+  SearchSource,
+} from '../../../../../../data/public';
+import { DataMain$, DataRefetch$ } from '../services/use_saved_search';
+import { DiscoverSearchSessionManager } from '../services/discover_search_session';
+
+/**
+ * This function returns an observable that's used to trigger data fetching
+ */
+export function getFetch$({
```

Comment:
As synced now. I think we could improve this a bit, by making this a hook `useFetch()`, which would return a type like the following:

```ts
{
  fetch$: Observable;
  done: () => void;
}
```

With `fetch$` being the current return value and `done` being a callback, that will internally do:

```
  refs.current.autoRefreshDone?.();
  refs.current.autoRefreshDone = undefined;
```

That way we also can move the ref completely into that hook itself.

Reply:
I've started working on this, and unfortunately we're currently calling this inside a hook, and not on the top-level. Thus we can't convert this into a hook, unless we'd also restructure the code in `useSavedSearch` a bit more. Given that we'd either need to call this then more often, or need to memorize this more, I feel the code quality and readability does actually not significantly increase by that suggestion anymore.

I'd thus suggest leaving it for now like it is and see if we can optimize that code in further refactorings better.